src/resolvers/mutation/delete-system-rolebinding/index.js

```
@@ -0,0 +1,50 @@
+import {
+  InvalidRoleError,
+  MissingRoleBindingError,
+  NoSystemAdminError
+} from "errors";
+import { startsWith } from "lodash";
+
+/*
+ * Delete a system role binding.
+ * @param {Object} parent The result of the parent resolver.
+ * @param {Object} args The graphql arguments.
+ * @param {Object} ctx The graphql context.
+ */
+export default async function deleteSystemRoleBinding(parent, args, ctx, info) {
+  // Pull out some args.
+  const { userId, role } = args;
+
+  // Throw error if system role not specified.
+  if (!startsWith(args.role, "SYSTEM")) throw new InvalidRoleError();
+
+  // Check if the RoleBinding already exists. If not, throw error.
+  const where = { role, user: { id: userId } };
+  const exists = await ctx.db.exists.RoleBinding(where);
+  if (!exists) throw new MissingRoleBindingError();
+
+  // Query for all SYSTEM_ADMIN RoleBindings across all users.
+  const systemAdmins = await ctx.db.query.roleBindings(
+    { where: { role } },
+    "{ role }"
+  );
+
+  // Throw error if there are less than 2 users with role SYSTEM_ADMIN
+  if (systemAdmins.length < 2) throw new NoSystemAdminError();
+
+  // Query for SYSTEM_ADMIN RoleBinding for the given user.
+  const roleBindings = await ctx.db.query.roleBindings(
+    { where: { role, user: { id: userId } } },
+    "{ role, id }"
+  );
```

@ianstanton could this and the `exists` query above possibly be combined? Maybe we could move this query up to replace it. Then we could throw the `MissingRoleBindingError` if `size(roleBindings) === 0`, and we could grab the `id` up front, rather than running a second query down here to grab it.

---

Updated